Packs/ThousandEyes/Integrations/ThousandEyes/ThousandEyes.py

```diff
@@ -0,0 +1,758 @@
+            json_data (dict): The dictionary to send in a 'POST' request. Default is None.
+            timeout: (float): Time (in seconds) for the client to wait to establish a connection before a timeout occurs.
+                                Default is 10.
+            resp_type (str): Determines which data format to return from the HTTP request. Other options are 'text',
+                             'content', 'xml' or 'response'. Use 'response' to return the full response object.
+                             Default is response.
+            test_module (bool): Boolean flag to check if function is called from test-module or not
+
+        Returns:
+            Either a Response Object or Dictionary, depending on the resp_type value
+        """
+
+        if test_module:
+            response = self._http_request(
+                method=method,
+                url_suffix=url_suffix,
+                resp_type=resp_type
+            )
+
+            return response
+
+        try:
+            while True:
+                response = self._http_request(
+                    method=method,
+                    url_suffix=url_suffix,
+                    params=params,
+                    data=data,
+                    json_data=json_data,
+                    timeout=timeout,
+                    resp_type=resp_type,
+                )
+
+                response_headers = response.headers
+                org_rate_limit_remaining = response_headers.get('x-organization-rate-limit-remaining')
+                org_rate_limit_reset = response_headers.get('x-organization-rate-limit-reset')
+
+                if response.status_code == 200 and int(org_rate_limit_remaining) > 0:
+                    return response
+
+                elif org_rate_limit_remaining == 0:
+                    current_timestamp = int(time.time())
+                    time.sleep(org_rate_limit_reset - current_timestamp)
+                    continue
+
+                else:
+                    status_code = response.status_code
+                    msg = response.text
+                    err_response = {
+                        "Status Code": status_code,
+                        "Err Message": msg
+                    }
+                    return err_response
+
+        except Exception as e:
+            raise Exception(e)
+
+
+def test_module_command(client: Client, *_) -> str:
+    """
+        Checks to see if the connection is working as expected or not
+
+    Args:
+        client (Client Object): Client object with request.
+
+    Returns:
+        'ok' if test successful.
+
+    Raises:
+        DemistoException: If test failed.
+    """
+
+    endpoint: str = '/status.json'
+    method: str = 'GET'
+
+    response = client.handle_rate_lmiit_and_make_request(
+        method=method,
+        url_suffix=endpoint,
+        test_module=True
+    )
+
+    if isinstance(response, dict):
+        raise DemistoException(f'Test module failed, {response}')
+
```

Comment:
I think the response can't be a dict (it's always a response) in case the command is `test-module`.

Reply:
Removed the 'dict' instance check.